normandy/control/static/control/js/actions/ControlActions.js

```diff
@@ -19,55 +17,41 @@ const BASE_API_URL = '/api/v1/recipe/';
 const API_REQUEST_SETTINGS = {
   credentials: 'include',
   headers: {
-    'X-CSRFToken': document.getElementsByTagName('html')[0].dataset.csrf
+    'X-CSRFToken': document.getElementsByTagName('html')[0].dataset.csrf,
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
   }
 };
 
 const apiRequestMap = {
   fetchAllRecipes(settings, getState) {
-    if (shouldFetchRecipes(getState())) {
-      return {
-        url: BASE_API_URL,
-        settings: {
-          method: 'get'
-        },
-        actionOnSuccess: recipesReceived,
-        errorNotification: 'Error fetching recipes.'
-      };
-    } else {
-      return null;
-    }
-  },
-
-  fetchSingleRecipe(recipeInfo) {
     return {
-      url: `${BASE_API_URL}${recipeInfo.recipeId}/`,
+      url: BASE_API_URL,
       settings: {
-        method: 'get'
+        method: 'GET'
       },
-      actionOnSuccess: singleRecipeReceived,
-      errorNotification: 'Error fetching recipe.'
+      errorNotification: 'Error fetching recipes.'
     };
   },
 
-  fetchSingleRevision(recipeInfo) {
+  fetchSingleRecipe(recipeInfo) {
+    let fetchUrl = (recipeInfo.recipeId ? `${BASE_API_URL}${recipeInfo.recipeId}/` : `/api/v1/recipe_version/${recipeInfo.revisionId}/`)
```

Comment:
it's weird to only use `BASE_API_URL` in one spot and a full URL in another. Also `BASE_API_URL` isn't actually the base API url, it's just the URL to the recipe API itself.

Also why is `fetchSingleRecipe` handling fetching two different types of objects?